external/llvm-project/mlir/lib/Dialect/XModel/Transforms/SelectTargets.cpp

```diff
@@ -87,6 +89,11 @@ struct XModelSelectTargetsPass
       if (targetKrn)
         func->setAttr(targetsTag, b.getArrayAttr(targetKrn));
       else {
+        if (targetArchs.size()) {
+          auto msg = Twine("Target not found for '") + func.getName() + "'\n";
```

Comment:
`func.emitError()` or `func.emitOpError()`, and never printing straight to `llvm::errs()`